src/server/rest.cpp

```diff
@@ -58,17 +63,28 @@ namespace ouroboros
 
 	std::pair<std::string, std::string> extract_group_name(const std::string& aURI)
 	{
-		struct slre_cap match[2];
-		slre_match(full_regex, aURI.c_str(), aURI.length(), match, 2, 0);
-
 		std::pair<std::string, std::string> result;
 		
-		//Copy group title from match to remove remaining characters
-		std::string groupTitle(match[0].ptr);
-		groupTitle.erase(groupTitle.begin()+match[0].len, groupTitle.end());
+		//Check if user is accessing field in root first
+		struct slre_cap match[1];
+		if(slre_match(root_field_regex, aURI.c_str(), aURI.length(), match, 1, 0) >= 0){
+			//Root group is "server"
+			result.first.assign("server");
```

Comment:
We may want to consider renaming the name of the root group to something that would be harder for users of our system to collide with... Heck, something weird, like _server would be enough, so long as we document it. We can even simply document that all things starting with an underscore are reserved for use by the system.